### ISSUE TITLE
TST's sidebar doubles keyboard input lag

### ISSUE THREAD

**Issue by pintassilgo** [piroor/treestyletab#3018]
I'm having a hard time trying to get exact steps to reproduce, but some things I have concluded so far:

- Visible Tree Style Tab sidebar doubles keyboard input lag (17ms → 34ms);
- Tabs overflow in sidebar is somewhat related.

### Context
I read someone complaining about FF input lag being higher than Chromium based browsers. I never noticed any issue in FF but decided to run an automated tool to compare. At first sight, yes, FF has twice the delay compared do Blink. But then I tried in a cleaner profile and it was as good as Blink. I did many many tests trying to isolate the cause (playing with prefs.js, toggling extensions, safe mode etc) and ultimately it's TST's fault.

### Tools

Automated tool to measure input lag: [Typometer](https://github.com/pavelfatin/typometer).

Textarea (dataUR)]:
data:text/html;charset=utf-8,%3Ctextarea%20rows%3D50%20cols%3D100%3E

Every test below was made using the same profile in the same session, with the same open tabs in a maximized window.

### First shot

TST in sidebar with overflow:
![image](https://user-images.githubusercontent.com/5483864/133884064-dfe96253-a553-463a-9a94-1165e24cd636.png)

History panel in sidebar (TST is enabled, just the sidebar is hidden):
![image](https://user-images.githubusercontent.com/5483864/133884365-5b77d20f-dc19-499b-9592-89b59a6e8d00.png)

TST in sidebar without overflow (less tabs visible by using a tab groups addon):
![image](https://user-images.githubusercontent.com/5483864/133884394-328002ea-6353-4344-b54d-9d6755eda261.png)

### Second shot

TST in sidebar with overflow:
![image](https://user-images.githubusercontent.com/5483864/133884442-2f067724-48ba-460c-8877-be2ad0cba2f0.png)

History panel in sidebar:
![image](https://user-images.githubusercontent.com/5483864/133884458-f1229a50-e0ac-4c2f-9b55-931d961986c5.png)

TST in sidebar without overflow:
![image](https://user-images.githubusercontent.com/5483864/133884470-1559b874-4482-4122-82ee-345a04cbb12d.png)

### Conclusion

The results are consistent.
- With hidden TST's sidebar, the input lag is low as it should be.
- With TST visible and without overflow it's a middle ground, input lag is low but with regular spikes probably due to some looping task performed by TST in background.
- With TST visible & overflow, the worst scenario, high and constant input lag.

Extra: tried another *vertical tabs* addon to compare ([ftt](https://addons.mozilla.org/en-US/firefox/addon/ftt/)) and it didn't affect the keyboard input lag, which remained low.

As I said, in real life I can't notice difference between 17ms and 34ms, it's not a big deal. But I fear that the true meaning of these results is that TST processing overhead is way higher than desired, even comparing to other vertical tabs addons (we all know TST is much more complex, but yet...).

**Comment by pintassilgo** [comment]
Now I'm guessing it's fault of Fx's layout render. The issue is also reproducible with Sidebery, another similar addon. I managed to narrow the issue by just cleaning sidebar's CSS using binary search. I reduced the entire CSS to just these instructions:

```css
.hidden-assets {
 position:absolute;
}
.Tab {
 height:0px;
}
```

Of course Sidebery's vertical tabbar becomes illegible that way, it's just about investigating the issue.

With the code above, input delay is still >30ms. But if I remove *any* of the two statements above (or both, removing the entire CSS from sidebar), then the input delay drops to <20ms.

I would like to report to Bugzilla, but it's too oddly to describe to the point of being taken seriously.
